test/runtime/jhh/lps_test.py

```diff
@@ -23,33 +32,51 @@ def run(cmd, env=None):
     else:
         proc = subprocess.run(cmd, text=True, check=True, env=env,
             stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
-    return proc
+    return proc.stdout
 
-def checkConfig():
+def skipif():
+    global skipReason
+
+    # Get the Chapel configuration
+    printchplenv.compute_all_values()
+    # strip the padding printchplenv puts on some of the keys
+    env = {k.strip():v for k,v in printchplenv.ENV_VALS.items()}
+
+    # Verify Chapel configuration
     # These tests only run on comm=ofi and slurm-srun
-    proc = run("printchplenv --simple")
-    for line in proc.stdout.splitlines():
-        (key, value) = line.split('=')
-        if key == 'CHPL_COMM' and value != 'ofi':
-            raise unittest.SkipTest("CHPL_COMM != ofi")
-        if key == 'CHPL_LAUNCHER' and value != 'slurm-srun':
-            raise unittest.SkipTest("CHPL_LAUNCHER != slurm-srun")
+    if env.get('CHPL_LAUNCHER', None) != 'slurm-srun':
+        skipReason = "CHPL_LAUNCHER != slurm-srun"
+        print(skipReason)
+        return
+    if env.get('CHPL_COMM', None) != 'ofi':
+        skipReason = "CHPL_COMM != ofi"
+        return
 
 class LocalePerSocket(unittest.TestCase):
     @classmethod
     def setUpClass(cls):
         global verbose
-        checkConfig()
-        cmd = ["sinfo", "--format=%X %Y %Z", "--noheader", "--exact"]
-        if 'CHPL_LAUNCHER_PARTITION' in os.environ:
-            cmd += ["--partition", os.environ['CHPL_LAUNCHER_PARTITION']]
+        if skipReason is not None:
+            return
+        cmd = ["sinfo", "--format=%X %Y %Z %i", "--noheader", "--exact"]
+        partition = os.environ.get('CHPL_LAUNCHER_PARTITION')
+        if partition is not None:
+            cmd += ["--partition", partition]
             if verbose:
-                print("Partition: ", os.environ['CHPL_LAUNCHER_PARTITION'])
-
-        proc = run(cmd)
-
-        (cls.sockets, cls.cores, cls.threads) = \
-            [int(i) for i in proc.stdout.splitlines()[0].split()]
+                print("Partition: ", partition)
+
+        output = runCmd(cmd)
+        if output.startswith("Invalid node format specification: i"):
+            # sinfo doesn't support %i (reservation name)
+            cmd[1] = "--format=%X %Y %Z"
```

Comment:
This check feels very fragile, can this be done in a separate function with "sinfo version". And then do this earlier so that 'cmd' is defined all together, makes the code a little easier to understand
```
if supports_reservation_name():
cmd = [...]
else
cmd = [...]
```